### ISSUE TITLE
Temp input changed metrics after update

### ISSUE THREAD

**Issue by electrofloat** [influxdata/telegraf#14646]
### Relevant telegraf.conf

```toml
[[inputs.temp]]
```

### Logs from Telegraf

```text
2024-01-29T21:14:59Z I! Starting Telegraf 1.29.3 brought to you by InfluxData the makers of InfluxDB
2024-01-29T21:14:59Z I! Loaded inputs: conntrack cpu disk diskio docker exec execd fail2ban internal interrupts kernel linux_cpu linux_sysctl_fs lvm mem net netstat nstat processes smart snmp swap system systemd_units temp
```

### System info

Telegraf, Ubuntu 22.04

### Steps to reproduce

Yesterday the telegraf package on Ubuntu from the influx repo was updated. I can't access the machine right now but looking at https://repos.influxdata.com/debian/packages/ it was probably updated to version 1.29.3-1. (because of the date: 2024-01-29 20:58:56+00:00 )

Since then the temperature metrics have changed.
First.. it seems the `acpitz` measurement randomly shows critical/max/input values.
Second.. there are new `_crit`, `_crit_alarm`, `_max`, `_input` values. It seems that now the `_input` values are the ones that show the actual temperature.

I've got hundreds of alert emails because of this.

### Expected behavior

I really don't know if it is a normal proceeding here to randomly change a metric. If it's not, then this is a bug.
If it is, please tell me how can I get notified about such changes _before_ it gets to the debian repo.

### Actual behavior

I think the metrics should not have changed.

**Comment by srebhan** [comment]
@electrofloat sorry for breaking your setup!!! It's probably my PR #14575 to blame. Can you please downgrade to v1.29.2 and help me to sort out the regression!? I would need a "copy" of the `temp*` files, the `name` file and the name of the `device` symlink target of `/sys/class/hwmon/*` directories. Furthermore, I would love to get the v1.29.2 outputs to make sure I fixed the regression...

**Comment by mbentley** [comment]
My apologies if this isn't the same cause but I came searching for the latest change and landed here.  I have a raspberry pi 3b+ where I am seeing a new additional metric which appears to be the critical temperature and not the sensor data being returned that shows up when upgraded to `1.29.3-1`.  It wouldn't be a big deal for me but with the critical temp not having anything to differentiate between it and the read temp, it triggered my temperature alert.

```
# 1.29.2-1
# telegraf --config /etc/telegraf/telegraf.conf --test --input-filter temp
2024-01-30T12:04:00Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T12:04:00Z I! Starting Telegraf 1.29.2 brought to you by InfluxData the makers of InfluxDB
2024-01-30T12:04:00Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-01-30T12:04:00Z I! Loaded inputs: temp
2024-01-30T12:04:00Z I! Loaded aggregators:
2024-01-30T12:04:00Z I! Loaded processors:
2024-01-30T12:04:00Z I! Loaded secretstores:
2024-01-30T12:04:00Z W! Outputs are not used in testing mode!
2024-01-30T12:04:00Z I! Tags enabled: host=rpi3bplus
> temp,host=rpi3bplus,sensor=cpu_thermal temp=52.078 1706616241000000000
```

```
# 1.29.3-1
# telegraf --config /etc/telegraf/telegraf.conf --test --input-filter temp
2024-01-30T12:00:34Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T12:00:34Z I! Starting Telegraf 1.29.3 brought to you by InfluxData the makers of InfluxDB
2024-01-30T12:00:34Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-
stores
2024-01-30T12:00:34Z I! Loaded inputs: temp
2024-01-30T12:00:34Z I! Loaded aggregators:
2024-01-30T12:00:34Z I! Loaded processors:
2024-01-30T12:00:34Z I! Loaded secretstores:
2024-01-30T12:00:34Z W! Outputs are not used in testing mode!
2024-01-30T12:00:34Z I! Tags enabled: host=rpi3bplus
> temp,host=rpi3bplus,sensor=cpu_thermal temp=49.926 1706616035000000000
> temp,host=rpi3bplus,sensor=cpu_thermal temp=90 1706616035000000000
```

```
# ls -l /sys/class/hwmon/*
lrwxrwxrwx 1 root root 0 Jan  3 06:16 /sys/class/hwmon/hwmon0 -> ../../devices/virtual/thermal/thermal_zone0/hwmon0/

# ls -l /sys/devices/virtual/thermal/thermal_zone0/hwmon0/
total 0
lrwxrwxrwx 1 root root    0 Jan 30 06:50 device -> ../../thermal_zone0/
-r--r--r-- 1 root root 4096 Jan  3 06:16 name
drwxr-xr-x 2 root root    0 Jan 30 07:08 power/
lrwxrwxrwx 1 root root    0 Jan 30 07:08 subsystem -> ../../../../../class/hwmon/
-r--r--r-- 1 root root 4096 Jan  3 06:16 temp1_crit
-r--r--r-- 1 root root 4096 Jan  3 06:16 temp1_input
-rw-r--r-- 1 root root 4096 Nov  9 19:25 uevent

# for TEMP in /sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp*_*; do echo $TEMP; cat $TEMP; done
/sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp1_crit
90000
/sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp1_input
49388
```

**Comment by mbentley** [comment]
Just for another data point, I also see on another system I just updated to `1.29.3-1` from the deb repos that the names have changed there as well which I assume would be undesirable for backwards compatibility where no configuration is being passed (same as my other system):

```
# 1.29.2-1
2024-01-30T12:31:25Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T12:31:25Z I! Starting Telegraf 1.29.2 brought to you by InfluxData the makers of InfluxDB
2024-01-30T12:31:25Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-01-30T12:31:25Z I! Loaded inputs: temp
2024-01-30T12:31:25Z I! Loaded aggregators:
2024-01-30T12:31:25Z I! Loaded processors:
2024-01-30T12:31:25Z I! Loaded secretstores:
2024-01-30T12:31:25Z W! Outputs are not used in testing mode!
2024-01-30T12:31:25Z I! Tags enabled: host=hermes
> temp,host=hermes,sensor=acpitz temp=27.8 1706617885000000000
> temp,host=hermes,sensor=nvme_composite temp=46.85 1706617885000000000
> temp,host=hermes,sensor=nvme_sensor_1 temp=46.85 1706617885000000000
> temp,host=hermes,sensor=nvme_sensor_2 temp=47.85 1706617885000000000
> temp,host=hermes,sensor=coretemp_package_id_0 temp=47 1706617885000000000
> temp,host=hermes,sensor=coretemp_core_0 temp=47 1706617885000000000
> temp,host=hermes,sensor=coretemp_core_1 temp=47 1706617885000000000
> temp,host=hermes,sensor=coretemp_core_2 temp=47 1706617885000000000
> temp,host=hermes,sensor=coretemp_core_3 temp=47 1706617885000000000
```

```
# 1.29.3-1
# telegraf --config /etc/telegraf/telegraf.conf --test --input-filter temp
2024-01-30T12:32:08Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T12:32:08Z I! Starting Telegraf 1.29.3 brought to you by InfluxData the makers of InfluxDB
2024-01-30T12:32:08Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-01-30T12:32:08Z I! Loaded inputs: temp
2024-01-30T12:32:08Z I! Loaded aggregators:
2024-01-30T12:32:08Z I! Loaded processors:
2024-01-30T12:32:08Z I! Loaded secretstores:
2024-01-30T12:32:08Z W! Outputs are not used in testing mode!
2024-01-30T12:32:08Z I! Tags enabled: host=hermes
2024-01-30T12:32:08Z W! [inputs.temp] Couldn't read temperature from "/sys/class/hwmon/hwmon3/temp1_input": read /sys/class/hwmon/hwmon3/temp1_input: no data available
> temp,host=hermes,sensor=acpitz temp=27.8 1706617928000000000
> temp,host=hermes,sensor=acpitz temp=110 1706617928000000000
> temp,host=hermes,sensor=nvme_composite_input temp=46.85 1706617928000000000
> temp,host=hermes,sensor=nvme_composite_alarm active=false 1706617928000000000
> temp,host=hermes,sensor=nvme_composite_crit temp=84.85 1706617928000000000
> temp,host=hermes,sensor=nvme_composite_max temp=84.85 1706617928000000000
> temp,host=hermes,sensor=nvme_composite_min temp=-273.15 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_1_input temp=46.85 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_1_max temp=65261.85 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_1_min temp=-273.15 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_2_input temp=52.85 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_2_max temp=65261.85 1706617928000000000
> temp,host=hermes,sensor=nvme_sensor_2_min temp=-273.15 1706617928000000000
> temp,host=hermes,sensor=coretemp_package_id_0_input temp=48 1706617928000000000
> temp,host=hermes,sensor=coretemp_package_id_0_crit temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_package_id_0_crit_alarm temp=0 1706617928000000000
> temp,host=hermes,sensor=coretemp_package_id_0_max temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_0_input temp=46 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_0_crit temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_0_crit_alarm temp=0 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_0_max temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_1_input temp=45 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_1_crit temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_1_crit_alarm temp=0 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_1_max temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_2_input temp=45 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_2_crit temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_2_crit_alarm temp=0 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_2_max temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_3_input temp=45 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_3_crit temp=105 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_3_crit_alarm temp=0 1706617928000000000
> temp,host=hermes,sensor=coretemp_core_3_max temp=105 1706617928000000000
```


```
# ls -l /sys/class/hwmon/*
lrwxrwxrwx 1 root root 0 Jan 17 16:35 /sys/class/hwmon/hwmon0 -> ../../devices/virtual/thermal/thermal_zone0/hwmon0/
lrwxrwxrwx 1 root root 0 Jan 17 16:35 /sys/class/hwmon/hwmon1 -> ../../devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/
lrwxrwxrwx 1 root root 0 Jan 17 16:35 /sys/class/hwmon/hwmon2 -> ../../devices/platform/coretemp.0/hwmon/hwmon2/
lrwxrwxrwx 1 root root 0 Jan 17 16:35 /sys/class/hwmon/hwmon3 -> ../../devices/virtual/thermal/thermal_zone1/hwmon3/
```

```
# ls -l /sys/devices/virtual/thermal/thermal_zone0/hwmon0/
total 0
lrwxrwxrwx 1 root root    0 Jan 17 16:35 device -> ../../thermal_zone0/
-r--r--r-- 1 root root 4096 Jan 17 16:35 name
drwxr-xr-x 2 root root    0 Jan 30 07:36 power/
lrwxrwxrwx 1 root root    0 Jan 17 16:35 subsystem -> ../../../../../class/hwmon/
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_input
-rw-r--r-- 1 root root 4096 Jan 17 16:35 uevent

# for TEMP in /sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp*_*; do echo $TEMP; cat $TEMP; done
/sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp1_crit
110000
/sys/devices/virtual/thermal/thermal_zone0/hwmon0/temp1_input
27800


# ls -l /sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/
total 0
lrwxrwxrwx 1 root root    0 Jan 17 16:35 device -> ../../nvme0/
-r--r--r-- 1 root root 4096 Jan 17 16:35 name
drwxr-xr-x 2 root root    0 Jan 30 07:38 power/
lrwxrwxrwx 1 root root    0 Jan 17 16:35 subsystem -> ../../../../../../../class/hwmon/
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_label
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp1_max
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp1_min
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_label
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp2_max
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp2_min
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_label
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp3_max
-rw-r--r-- 1 root root 4096 Jan 17 16:35 temp3_min
-rw-r--r-- 1 root root 4096 Jan 17 16:35 uevent

# for TEMP in /sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp*_*; do echo $TEMP; cat $TEMP; done
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_alarm
0
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_crit
84850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_input
46850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_label
Composite
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_max
84850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp1_min
-273150
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp2_input
46850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp2_label
Sensor 1
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp2_max
65261850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp2_min
-273150
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp3_input
48850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp3_label
Sensor 2
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp3_max
65261850
/sys/devices/pci0000:00/0000:00:1d.0/0000:04:00.0/nvme/nvme0/hwmon1/temp3_min
-273150


# ls -l /sys/devices/platform/coretemp.0/hwmon/hwmon2/
total 0
lrwxrwxrwx 1 root root    0 Jan 17 16:35 device -> ../../../coretemp.0/
-r--r--r-- 1 root root 4096 Jan 17 16:35 name
drwxr-xr-x 2 root root    0 Jan 30 07:39 power/
lrwxrwxrwx 1 root root    0 Jan 17 16:35 subsystem -> ../../../../../class/hwmon/
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_crit_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_label
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_max
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_crit_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_label
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp2_max
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_crit_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_label
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp3_max
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp4_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp4_crit_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp4_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp4_label
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp4_max
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp5_crit
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp5_crit_alarm
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp5_input
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp5_label
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp5_max
-rw-r--r-- 1 root root 4096 Jan 17 16:35 uevent

# for TEMP in /sys/devices/platform/coretemp.0/hwmon/hwmon2/temp*_*; do echo $TEMP; cat $TEMP; done
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp1_crit
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp1_crit_alarm
0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp1_input
45000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp1_label
Package id 0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp1_max
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp2_crit
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp2_crit_alarm
0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp2_input
42000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp2_label
Core 0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp2_max
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp3_crit
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp3_crit_alarm
0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp3_input
42000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp3_label
Core 1
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp3_max
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp4_crit
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp4_crit_alarm
0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp4_input
42000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp4_label
Core 2
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp4_max
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp5_crit
105000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp5_crit_alarm
0
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp5_input
43000
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp5_label
Core 3
/sys/devices/platform/coretemp.0/hwmon/hwmon2/temp5_max
105000


# ls -l /sys/devices/virtual/thermal/thermal_zone1/hwmon3/
total 0
lrwxrwxrwx 1 root root    0 Jan 17 16:35 device -> ../../thermal_zone1/
-r--r--r-- 1 root root 4096 Jan 17 16:35 name
drwxr-xr-x 2 root root    0 Jan 30 07:39 power/
lrwxrwxrwx 1 root root    0 Jan 30 07:39 subsystem -> ../../../../../class/hwmon/
-r--r--r-- 1 root root 4096 Jan 17 16:35 temp1_input
-rw-r--r-- 1 root root 4096 Jan 17 16:35 uevent

# for TEMP in /sys/devices/virtual/thermal/thermal_zone1/hwmon3/temp*_*; do echo $TEMP; cat $TEMP; done
/sys/devices/virtual/thermal/thermal_zone1/hwmon3/temp1_input
cat: /sys/devices/virtual/thermal/thermal_zone1/hwmon3/temp1_input: No data available
```

**Comment by srebhan** [comment]
@electrofloat and @mbentley again, please accept my apologies for causing inconvenience! I put up a potential fix in PR #14649 and would like to ask you to test the binaries there, available as soon as CI finished the tests. Please let me know if this fixes your issues!

**Comment by electrofloat** [comment]
Hey, no worries and thanks for the quick reaction.

I won't add my lists then, since it's basically the same as @mbentley's.

I've installed the .deb from the PR (https://output.circle-artifacts.com/output/job/48be69f2-7a14-436e-a627-d6a2a4d39ef0/artifacts/0/build/dist/telegraf_1.30.0~d8f36c37-0_amd64.deb) and it indeed seems to solve the issue:

```
# telegraf --config /etc/telegraf/telegraf.d/telegraf.conf --test --input-filter temp
2024-01-30T16:57:22Z I! Loading config: /etc/telegraf/telegraf.d/telegraf.conf
2024-01-30T16:57:22Z I! Starting Telegraf 1.30.0-d8f36c37 brought to you by InfluxData the makers of InfluxDB
2024-01-30T16:57:22Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 61 outputs, 6 secret-stores
2024-01-30T16:57:22Z I! Loaded inputs: temp
2024-01-30T16:57:22Z I! Loaded aggregators:
2024-01-30T16:57:22Z I! Loaded processors:
2024-01-30T16:57:22Z I! Loaded secretstores:
2024-01-30T16:57:22Z W! Outputs are not used in testing mode!
2024-01-30T16:57:22Z I! Tags enabled: host=ww
> temp,host=ww,sensor=acpitz temp=27.8 1706633842000000000
> temp,host=ww,sensor=nvme_composite temp=40.85 1706633842000000000
> temp,host=ww,sensor=coretemp_core_23 temp=45 1706633842000000000
> temp,host=ww,sensor=coretemp_package_id_0 temp=48 1706633842000000000
> temp,host=ww,sensor=coretemp_core_0 temp=48 1706633842000000000
> temp,host=ww,sensor=coretemp_core_16 temp=44 1706633842000000000
> temp,host=ww,sensor=coretemp_core_17 temp=44 1706633842000000000
> temp,host=ww,sensor=coretemp_core_18 temp=44 1706633842000000000
> temp,host=ww,sensor=coretemp_core_19 temp=44 1706633842000000000
> temp,host=ww,sensor=coretemp_core_20 temp=45 1706633842000000000
> temp,host=ww,sensor=coretemp_core_21 temp=45 1706633842000000000
> temp,host=ww,sensor=coretemp_core_22 temp=45 1706633842000000000
```

no more `_max`, `_input`, etc.

**Comment by srebhan** [comment]
Thanks for the confirmation @electrofloat! @mbentley can you please also confirm that the PR fixes the issue for you!?

**Comment by srebhan** [comment]
Oh and @electrofloat is `acpitz` behaving correctly as you mentioned it has random content!?!?

**Comment by mbentley** [comment]
Quick check for me and it looks like it is back to the expected outputs:

On my raspberry pi:
```
# telegraf --config /etc/telegraf/telegraf.conf --test --input-filter temp
2024-01-30T19:26:23Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T19:26:23Z I! Starting Telegraf 1.30.0-d8f36c37 brought to you by InfluxData the makers of InfluxDB
2024-01-30T19:26:23Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 61 outputs, 6 secret-stores
2024-01-30T19:26:23Z I! Loaded inputs: temp
2024-01-30T19:26:23Z I! Loaded aggregators:
2024-01-30T19:26:23Z I! Loaded processors:
2024-01-30T19:26:23Z I! Loaded secretstores:
2024-01-30T19:26:23Z W! Outputs are not used in testing mode!
2024-01-30T19:26:23Z I! Tags enabled: host=rpi3bplus
> temp,host=rpi3bplus,sensor=cpu_thermal temp=50.464 1706642784000000000
```


on my amd64 system:
```
# telegraf --config /etc/telegraf/telegraf.conf --test --input-filter temp
2024-01-30T19:25:06Z I! Loading config: /etc/telegraf/telegraf.conf
2024-01-30T19:25:06Z I! Starting Telegraf 1.30.0-d8f36c37 brought to you by InfluxData the makers of InfluxDB
2024-01-30T19:25:06Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 61 outputs, 6 secret-stores
2024-01-30T19:25:06Z I! Loaded inputs: temp
2024-01-30T19:25:06Z I! Loaded aggregators:
2024-01-30T19:25:06Z I! Loaded processors:
2024-01-30T19:25:06Z I! Loaded secretstores:
2024-01-30T19:25:06Z W! Outputs are not used in testing mode!
2024-01-30T19:25:06Z I! Tags enabled: host=hermes
2024-01-30T19:25:06Z W! [inputs.temp] Couldn't read temperature from "/sys/class/hwmon/hwmon3/temp1_input": read /sys/class/hwmon/hwmon3/temp1_input: no data available
> temp,host=hermes,sensor=acpitz temp=27.8 1706642706000000000
> temp,host=hermes,sensor=nvme_composite temp=46.85 1706642706000000000
> temp,host=hermes,sensor=nvme_sensor_1 temp=46.85 1706642706000000000
> temp,host=hermes,sensor=nvme_sensor_2 temp=51.85 1706642706000000000
> temp,host=hermes,sensor=coretemp_package_id_0 temp=46 1706642706000000000
> temp,host=hermes,sensor=coretemp_core_0 temp=45 1706642706000000000
> temp,host=hermes,sensor=coretemp_core_1 temp=45 1706642706000000000
> temp,host=hermes,sensor=coretemp_core_2 temp=45 1706642706000000000
> temp,host=hermes,sensor=coretemp_core_3 temp=46 1706642706000000000
```

**Comment by akordowski** [comment]
> Thank you both for confirming. I am waiting on one other issue and will do a release with both shortly after.

I testedt the latest docker image and the issue is still there. Are the changes released yet?

**Comment by powersj** [comment]
> I tested the latest docker image and the issue is still occurring. Are the changes released yet?

Yes - it was fixed and released in 1.29.4: https://github.com/docker-library/official-images/pull/16160

Assuming you have upgraded to 1.29.4, if you are having issue please file a new issue with the files found on your system similar to [this post](https://github.com/influxdata/telegraf/issues/14646#issuecomment-1916765598) so we can reproduce.

**Comment by akordowski** [comment]
The error was on my end, I had an older image with the tag `latest`. I pulled the new one and it work as expected. Sorry for the inconvenience.
